src/bridges/nft-basic/NftVault.sol

```diff
@@ -0,0 +1,147 @@
+     *                              corresponding to the returned virtual asset id
+     * @param _auxData - corresponds to the Ethereum address id in the AddressRegistry.sol for withdrawals
+     * @return outputValueA - 1 VIRTUAL asset (Deposit) or 0 ETH (Withdrawal)
+     *
+     */
+
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 _totalInputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override (BridgeBase)
+        onlyRollup
+        returns (uint256 outputValueA, uint256 outputValueB, bool isAsync)
+    {
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidInputA();
+        if (
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidOutputA();
+        if (_totalInputValue != 1) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.ETH
+                && _outputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+        ) {
+            return (1, 0, false);
+        } else if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+                && _outputAssetA.assetType == AztecTypes.AztecAssetType.ETH
+        ) {
+            NftAsset memory token = nftAssets[_inputAssetA.id];
+            if (token.collection == address(0x0)) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            address to = REGISTRY.addresses(_auxData);
+            if (to == address(0x0)) {
+                revert ErrorLib.InvalidAuxData();
+            }
+            delete nftAssets[_inputAssetA.id];
+            IERC721(token.collection).transferFrom(address(this), to, token.tokenId);
+            emit NftWithdraw(_inputAssetA.id, token.collection, token.tokenId);
+            return (0, 0, false);
+        } else if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+                && _outputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+        ) {
+            NftAsset memory token = nftAssets[_inputAssetA.id];
+            if (token.collection == address(0x0)) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            address to = REGISTRY.addresses(_auxData);
+            delete nftAssets[_inputAssetA.id];
+
+            IERC721(token.collection).approve(to, token.tokenId);
+            NftVault(to).transferFromAndMatch(_interactionNonce, token.collection, token.tokenId);
```

Comment:
Intuitively, the action here is a withdraw followed by a deposit. So it should probably emit the same events.